EX03/app/src/main/java/com/example/ex03/NotifyIntentService.java

```diff
@@ -0,0 +1,58 @@
+package com.example.ex03;
+
+import android.app.AlarmManager;
+import android.app.IntentService;
+import android.app.PendingIntent;
+import android.content.Intent;
+import android.content.Context;
+
+public class NotifyIntentService extends IntentService {
+    private static final String ACTION_BUMP_QUOTE = "com.example.exercise_03.action.BUMP_QUOTE";
+
+
+    public NotifyIntentService() {
+        super("NotifyIntentService");
+    }
+
+    /**
+     * Starts this service to perform action Foo with the given parameters. If
+     * the service is already performing a task this action will be queued.
+     *
+     * @see IntentService
+     */
+
+    private void startAlarm() {
+        AlarmManager alarmManager = (AlarmManager)this.getSystemService(this.ALARM_SERVICE);
+        long when = System.currentTimeMillis();
+        Intent intent = new Intent(this, NotificationReceiver.class);
+        intent.addFlags(Intent.FLAG_RECEIVER_FOREGROUND);
+        PendingIntent pendingIntent = PendingIntent.getBroadcast(this, NotificationReceiver.REQUEST_CODE, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+        alarmManager.setRepeating(AlarmManager.RTC_WAKEUP, when, (AlarmManager.INTERVAL_FIFTEEN_MINUTES / 3), pendingIntent);
```

Comment:
when is not very indicative...
Would've move (AlarmManager.INTERVAL_FIFTEEN_MINUTES / 3) into a variable explaining what it is